### PR TITLE
0.9.12 - heskenn

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
cc @hroncok , @rgommers, @mgorny @jeanlaroche  I'll leave that branch alive for a few days before merging so that you can test it.

**Comment by hroncok** [comment]
Testing...

File: Changelog [hunk 1]
```diff
@@ -1,3 +1,27 @@
+2021-07-06 Serge Guelton <serge.guelton@telecom-bretagne.eu>
+
+	* Bump gast and Beniget requirements to support python 3.10
+
+	* Bump xsimd to 7.50
```
Comment (hroncok) [inline Changelog:5]:
```suggestion
	* Bump xsimd to 7.5.0
```

**Comment by hroncok** [comment]
The Fedora build fails tests on 32bit arches and s390x.

I'll report individual issues.

The 32bit one is https://github.com/serge-sans-paille/pythran/issues/1834

**Comment by hroncok** [comment]
The s390x failures are "Too many open files".

```
.................................s...................................... [  2%]
................................................s....................... [  4%]
........................................................................ [  6%]
........................................................................ [  8%]
.................................s...................................s.. [ 10%]
......s...................s............................................. [ 12%]
...........................................s..........................s. [ 14%]
...................................s....s....s.....s.s.................. [ 16%]
.............................................................s..s....... [ 18%]
..s.................s...........s........s.............................. [ 20%]
........................................................................ [ 22%]
................................................s....................... [ 24%]
........................................................................ [ 27%]
........................................................................ [ 29%]
.......................ss.....................sss....................... [ 31%]
........................................................................ [ 33%]
........................................................................ [ 35%]
........................................................................ [ 37%]
.....s.................................sss........................ss.... [ 39%]
........................................................................ [ 41%]
........................................................................ [ 43%]
........................................................................ [ 45%]
...........................s............................................ [ 47%]
........................................................................ [ 49%]
...............................ss....................................... [ 51%]
...................................................s.................... [ 54%]
........................................................................ [ 56%]
........................................................................ [ 58%]
........................................................................ [ 60%]
........................................................................ [ 62%]
........................................................................ [ 64%]
........................................................................ [ 66%]
......................................FFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 68%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 70%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 72%]
FFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFF.FFFF [ 74%]
FFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFF.FFFFFFFFFFF [ 76%]
FFFF.FF........FFFFFF....F........F..FFFFFFFF.FF.FFF.F.FFFFFFFFFFFFFFFFF [ 79%]
FsFFF.FFFFFF.FFFFFFFFFFFFFFFFFFFFEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE [ 81%]
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEFFFF [ 83%]
FFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFF [ 85%]
FFFFFFFFFFFFFF..FFFFFFFFFF...........F.FFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFF [ 87%]
FFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFF.FFFFFFss.........FFF..FFFFFFFFFFsFFFF [ 89%]
F..FF.........ssFFFFFFFFFFFFF.FEEEEEEEEEEEE..FFFFFFFFFFFFFFFFFFFFFFFFFFF [ 91%]
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF [ 93%]
FFFFFFFFFFFF............................................................ [ 95%]
........................................................................ [ 97%]
........................................................................ [ 99%]
.......                                                                  [100%]
```

I wonder whether reducing the parallelism of the tests would help.

**Comment by rgommers** [comment]
Tested with SciPy master, LGTM

**Comment by mgorny** [comment]
The weather's become cloudy and not-so-damn-hot, so let's try heating the climate with Pythran's test suite ;-).

**Comment by hroncok** [comment]
> I wonder whether reducing the parallelism of the tests would help.

Nope, running with pytest-xdist `-n 1` does not help :/

**Comment by mgorny** [comment]
I can confirm that the tests pass on Gentoo/amd64.

**Comment by hroncok** [comment]
> > I wonder whether reducing the parallelism of the tests would help.
> 
> Nope, running with pytest-xdist `-n 1` does not help :/

Quite the contrary, running the tests with more parallel runners hides the problem and running it with just one makes the problem appear on other architectures as well. It was just a coincidence that the s390x runner only had 2 or 3 available cores but all the other architectures had more, so `-n auto` made it appear as s390x related.

It seems that the more test are run in one process, the more open files. Is it possible that pythran leaks open files somehow?

In Fedora, I can work around the failures by setting higher `ulimit -n`, but it seems suspicious. Shall I open an issue?

**Comment by serge-sans-paille** [comment]
> In Fedora, I can work around the failures by setting higher `ulimit -n`, but it seems suspicious. Shall I open an issue?

Please do. It's good that you have a workaround so that it's not blocking, but leaking open files doesn't look great.

**Comment by hroncok** [comment]
https://github.com/serge-sans-paille/pythran/issues/1835

**Comment by rgommers** [comment]
@serge-sans-paille can we add `networkx < 2.6` to `requirements.txt` here? That would address the problem in https://github.com/scipy/scipy/issues/14335#issuecomment-877010470.

**Comment by serge-sans-paille** [comment]
I'll rebase upon https://github.com/serge-sans-paille/pythran/pull/1839 instead

**Comment by serge-sans-paille** [comment]
I've updated that branch / PR with the fixes for the issues spotted by @rgommers and @hroncok , including a cleanup of the dependencies (networkx/six/decorator are no longer used). Feedback welcome!

**Comment by serge-sans-paille** [comment]
I'll merge that one in a few hours if nobody raises a red flag.

**Pull request merged**